feat: same titles for sent email groups

ihor-romaniuk opened the pull request:
**Reopened Pull Request based on** https://github.com/openedx/frontend-app-communications/pull/205
----

### Description
this PR proposes making the display of mail recipients the same for all places (checkbox labels, **Send email** alert modal, **Scheduled emails** table with **View Modal**, **Email Task History**)

### Steps to reproduce
click on "Staff/Administrators" checkbox -> complete the form (subject, body) -> options:

- to see recipients titles in "Caution" modal, click on "Send email" button
- same if you schedule day and time and click on "Schedule email" button
(in this modal you can see the difference between recipients titles and selected checkbox label)

-> click "Continue" button -> wait for the "Scheduled emails" table to load and also compare titles in "Send to" column

### Screenshots before
**Example with selected "Staff/Administrators"**

|  <img width="1686" height="862" alt="image" src="https://github.com/user-attachments/assets/5680574e-dab9-4669-ad07-b0ad537884ef" /> |  <img width="1623" height="1004" alt="image" src="https://github.com/user-attachments/assets/812832a2-03db-488a-9fc7-3dc1122c9739" /> |  <img width="1047" height="445" alt="image" src="https://github.com/user-attachments/assets/be3547eb-fdae-44d7-93e9-6e4fe25d80d5" /> |
|---|---|---|

**Example with selected "All Learners"**

|  <img width="976" height="592" alt="image" src="https://github.com/user-attachments/assets/e195762f-ba57-4854-ad1a-1986ffb0a963" /> |  <img width="713" height="459" alt="image" src="https://github.com/user-attachments/assets/281b7b84-f769-4164-b301-3172a17b16dc" /> |  <img width="1139" height="367" alt="image" src="https://github.com/user-attachments/assets/1d1ac974-c486-4d3a-b980-703bafc85d8a" /> |
|---|---|---|

### Screenshots after

|  <img width="2192" height="1102" alt="image" src="https://github.com/user-attachments/assets/8f6ea576-e5fe-4aab-b4b8-a00daa20faa2" /> |  <img width="1288" height="1026" alt="image" src="https://github.com/user-attachments/assets/11c35ffe-945b-48f2-bf92-17759c83dd41" /> | 
|---|---|

|  <img width="1868" height="602" alt="image" src="https://github.com/user-attachments/assets/1baf9dab-bd25-4e93-9aa0-9c9f23c623b9" /> |  <img width="2216" height="1426" alt="image" src="https://github.com/user-attachments/assets/edd31237-93d6-4c7a-9098-a2e3eacbaca0" /> |  <img width="2110" height="1204" alt="image" src="https://github.com/user-attachments/assets/05093dc5-3a87-4e85-9f8b-12f1b3a94ae0" /> |
|---|---|---|

Commit:
feat: same titles for sent email groups

Commit:
feat: aligned the checkbox labels with the BE

ihor-romaniuk committed:
fix: refactor the code

ihor-romaniuk committed:
fix: add fall back to the recipient key

Co-authored-by: Braden MacDonald <mail@bradenm.com>

ihor-romaniuk committed:
fix: lint error

brian-smith-tcril reviewed (approved):
I have not tested this but the code LGTM.

I left one comment with a question about a CSS class being added.

ihor-romaniuk committed:
fix: remove extra class name

ihor-romaniuk commented on src/components/bulk-email-tool/bulk-email-task-manager/ViewEmailModal.jsx:
```diff
@@ -34,7 +33,7 @@ function ViewEmailModal({
             </div>
             <div className="d-flex flex-row">
-              <p>{intl.formatMessage(messages.modalMessageSentTo)}</p>
+              <p className="flex-shrink-0">{intl.formatMessage(messages.modalMessageSentTo)}</p>
```
I have re-checked it and we can revert this class. Thanks
<img width="1727" height="864" alt="image" src="https://github.com/user-attachments/assets/a7f41aaf-aa03-4366-b6f9-1de09bf8256d" />

ihor-romaniuk committed:
chore: update package-lock.json

peterkulko reviewed (changes requested):
@ihor-romaniuk thanks for the PR! I left a minor comment and one question about the `package-lock.json` changes.

ihor-romaniuk commented on package-lock.json:
Yep - those changes were required. I only updated `package-lock.json` because it was out of date, and the PR checks were failing because of it. No functional changes intended there.

ihor-romaniuk committed:
fix: simple code improvements

ihor-romaniuk committed:
revert changes in package-lock file

peterkulko reviewed (approved):
LGTM!

arbrandes reviewed (changes requested):
Ok, some stuff Claude caught that I agree warrants addressing.  Only the first one is an actual bug, though:

### ViewEmailModal for scheduled emails still shows raw identifiers

In [`BulkEmailScheduledEmailsTable.jsx`](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L58-L71), the `handleViewEmail` function passes `row.original.targets` to the ViewEmailModal:

```jsx
sent_to: row.original.targets,
```

`targets` is the raw joined string (e.g., `"learners"` or `"staff"`), not the display-friendly `targetsText` (e.g., `"All learners"` or `"Staff and instructors"`). The table column was correctly switched to use `targetsText` at [line 160](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L160), but the View modal was missed.

This means: clicking "View" on a scheduled email will display "learners" in the "Sent to" field while the table row next to it says "All learners". This partially defeats the PR's purpose of making display names consistent across all views.

The fix is straightforward - use `targetsText` instead of `targets` on that line.

### Changed i18n message IDs for "Myself" label

The "Myself" label text didn't change, but its message ID was renamed from `bulk.email.form.recipients.myself` to `bulk.email.recipients.myself.label`. This means any existing translations for that string will be lost and fall back to the English default until re-translated.

For "Staff/Administrators" and "All Learners", new IDs make sense since the English text actually changed. For "Myself", keeping the old message ID (or extracting it to the shared file under the same ID) would preserve existing translations.

This may or may not matter depending on how actively translations are maintained for this MFE, but it's worth being aware of.

### Nit: unclear variable name

In [`BulkEmailScheduledEmailsTable.jsx` line 33](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L33):

```jsx
targetsText: email.courseEmail.targets.map((mess) => getDisplayTextFromRecipient(intl, mess)).join(', '),
```

The parameter `mess` reads like it might mean "message", but these items are recipient identifiers. Something like `recipient` or `target` would be clearer.

ihor-romaniuk committed:
fix: address review comments for recipient display names

- Use targetsText in ViewEmailModal so scheduled emails show
  display-friendly names instead of raw identifiers
- Restore original i18n message ID for "Myself" to preserve
  existing translations
- Rename unclear `mess` parameter to `recipient`

ihor-romaniuk commented:
> Ok, some stuff Claude caught that I agree warrants addressing. Only the first one is an actual bug, though:
> 
> ### ViewEmailModal for scheduled emails still shows raw identifiers
> In [`BulkEmailScheduledEmailsTable.jsx`](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L58-L71), the `handleViewEmail` function passes `row.original.targets` to the ViewEmailModal:
> 
> ```js
> sent_to: row.original.targets,
> ```
> 
> `targets` is the raw joined string (e.g., `"learners"` or `"staff"`), not the display-friendly `targetsText` (e.g., `"All learners"` or `"Staff and instructors"`). The table column was correctly switched to use `targetsText` at [line 160](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L160), but the View modal was missed.
> 
> This means: clicking "View" on a scheduled email will display "learners" in the "Sent to" field while the table row next to it says "All learners". This partially defeats the PR's purpose of making display names consistent across all views.
> 
> The fix is straightforward - use `targetsText` instead of `targets` on that line.
> 
> ### Changed i18n message IDs for "Myself" label
> The "Myself" label text didn't change, but its message ID was renamed from `bulk.email.form.recipients.myself` to `bulk.email.recipients.myself.label`. This means any existing translations for that string will be lost and fall back to the English default until re-translated.
> 
> For "Staff/Administrators" and "All Learners", new IDs make sense since the English text actually changed. For "Myself", keeping the old message ID (or extracting it to the shared file under the same ID) would preserve existing translations.
> 
> This may or may not matter depending on how actively translations are maintained for this MFE, but it's worth being aware of.
> 
> ### Nit: unclear variable name
> In [`BulkEmailScheduledEmailsTable.jsx` line 33](https://github.com/openedx/frontend-app-communications/blob/548471563136d81e5697f197657ec230ba0cf844/src/components/bulk-email-tool/bulk-email-task-manager/bulk-email-scheduled-emails-table/BulkEmailScheduledEmailsTable.jsx#L33):
> 
> ```js
> targetsText: email.courseEmail.targets.map((mess) => getDisplayTextFromRecipient(intl, mess)).join(', '),
> ```
> 
> The parameter `mess` reads like it might mean "message", but these items are recipient identifiers. Something like `recipient` or `target` would be clearer.

@arbrandes Thanks for the review! All three items are addressed in the latest commit:

- **ViewEmailModal bug** - fixed `sent_to` to use `targetsText` instead of raw `targets`, so the View modal now shows the same display-friendly names as the table.
- **"Myself" i18n ID** - restored the original `bulk.email.form.recipients.myself` message ID to preserve existing translations (since the default text didn't change).
- **Variable naming** - renamed mess to recipient for clarity.

Please re-check when you get a chance, and go forward if everything looks good.

arbrandes commented:
@ihor-romaniuk, almost there!  All issues address, thank you - but package-lock.json is out of sync with package.json, so `npm ci` doesn't work.  Mind fixing it up?

ihor-romaniuk committed:
fix: update package-lock file

ihor-romaniuk commented:
@arbrandes it looks like the package-lock file was not updated to use node 24.

I've rebuild the package-lock.json file and now all CI checkers pass.

arbrandes reviewed (approved):
Looks great, thanks!
